File: 0003-longest-substring-without-repeating-characters/0003-longest-substring-without-repeating-characters.cpp
class Solution {
public:
    int lengthOfLongestSubstring(string s) {
        unordered_set<char>st;
        int l=0,maxi=0;
        for(int i=0;i<s.size();i++){
            while(st.count(s[i])){
                st.erase(s[l]);
                l++;
            }
            st.insert(s[i]);
            maxi=max(maxi,i-l+1);
        }
        return maxi;
    }
};

File: 0003-longest-substring-without-repeating-characters/0003-longest-substring-without-repeating-characters.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        l=0
        seen={}
        maxi=0
        for i,curr in enumerate(s):
            if curr in seen:
                l=max(l,seen[curr]+1)
            maxi=max(maxi,i-l+1)
            seen[curr]=i
        return maxi